python/ray/train/base_trainer.py

```diff
@@ -347,6 +338,27 @@ def as_trainable(self) -> Type[Trainable]:
         scaling_config = self.scaling_config
 
         def train_func(config, checkpoint_dir=None):
+            # TODO(amogkam): Remove this warning after _max_cpu_fraction_per_node is no
+            #  longer experimental.
+            if self.datasets and not self.scaling_config._max_cpu_fraction_per_node:
+                if (
+                    ray.available_resources().get("CPU", 0)
+                    / ray.cluster_resources().get("CPU", 0)
+                    < 0.2
+                ):
+                    warnings.warn(
```

Comment:
Please revert/update this to:

Instantiating this Trainer leaves less than 20% of CPUs in this cluster for Dataset execution. To avoid this, it is recommended to explicitly reserve at least 20% of node CPUs for Dataset execution by setting `_max_cpu_fraction_per_node = 0.8` in the Trainer `scaling_config`. Not doing so can lead to resource contention or hangs. See https://docs.ray.io/en/master/data/key-concepts.html#example-datasets-in-tune for more info.

Reply:
updated